osu.Server.Spectator/Entities/EntityStore.cs

```diff
@@ -24,6 +26,33 @@ public sealed class EntityStore<T>
 
         private string statsDPrefix => $"entities.{typeof(T).Name}";
 
+        private bool acceptingNewEntities = true;
+
+        public EntityStore(GracefulShutdownManager shutdownManager, IHostApplicationLifetime applicationLifetime)
+            : this()
+        {
+            // Registration is required here to ensure `ApplicationStopping` triggers the shutdown wait before SignalR disconnects clients.
+            // Hope we can find a better way to make this work in the future.
```

Comment:
Not sure I see anything too offensive about doing this registration where it is. Where were you looking to put it?

Reply:
I don't understand this. You can inject dependencies into dependencies. So you could add a `GracefulShutdownManager(IHostApplicationLifetime applicationLifetime)` ctor afaik, which seems more appropriate?

Reply:
Give it a try and you'll see why I had to structure as it is.

Reply:
Has been resolved with new knowledge of how to make `ApplicationStopping` arrive in the correct order.